copy package.json to staging directory

danfiedler-msft opened the pull request.

gfs reviewed (changes requested):
I don't think this is the correct approach. The package.json is required as a component of the vsix and should be read out of it. I think the error actually means that it isn't finding the vsix to publish.

Additionally, we've found that its necessary to manually mark pre-release versions with vsce with a special flag, or else they will clobber all stable release versions.

See https://github.com/microsoft/DevSkim/blob/bd092d93005d6ac38a101bf4505e12fd09cba174/Pipelines/vscode/devskim-vscode-release.yml#L204-L219 for previous publish action using package path and switching on release/pre-release.

danfiedler-msft commented:
> I don't think this is the correct approach. The package.json is required as a component of the vsix and should be read out of it. I think the error actually means that it isn't finding the vsix to publish.

Updated the publish step to bring back release vs. prerelease. Added a step to list the contents of the staging directory. 

I know from the signing step the VSIX exists prior to the move from binaries to staging. 

> Signed file: D:\a\_work\1\b\Unsigned_Plugin\vscode-devskim-1.0.45.vsix

I think the move should work so it will be surprising if the VSIX isn't in the expected spot for publishing.
